x/sync/manager.go

```diff
@@ -49,14 +49,14 @@ const (
 // [localRootID] is the ID of the root of this range in our database.
 // If we have no local root for this range, [localRootID] is ids.Empty.
 type workItem struct {
-	start       []byte
-	end         []byte
+	start       merkledb.Maybe[[]byte]
```

Comment:
Would it make sense to define `Maybe` in a shared types package rather than just `merkledb`?

Reply:
I can do that in a follow up PR -- this one is already kind of huge